core/client/fs/src/main/java/alluxio/master/MasterHealthCheckClient.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master;
+
+import alluxio.AlluxioURI;
+import alluxio.HealthCheckClient;
+import alluxio.client.file.FileSystem;
+import alluxio.util.ShellUtils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetSocketAddress;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * MasterHealthCheckClient check whether Alluxio master is serving RPC request and
+ * if the AlluxioMaster process is running in all master hosts.
+ */
+public class MasterHealthCheckClient implements HealthCheckClient {
+
+  private static final Logger LOG = LoggerFactory.getLogger(MasterHealthCheckClient.class);
+
+  private Thread mMasterServingThread;
+  private Thread mMastersProcessCheckThread;
+  private ScheduledExecutorService mExecutorService = Executors.newScheduledThreadPool(2);
+
+  /**
+   * Creates a master health check client.
+   */
+  public MasterHealthCheckClient() {
+    mMasterServingThread = new Thread(() -> {
```

Comment:
The `ScheduledExecutorService` handles the threading, so I think we can just use regular `Runnables` here

Reply:
Done